acknowledgements.rst

```diff
@@ -14,5 +14,7 @@ Xilinx, Inc.
 
 NXP Semiconductors
 
+Cavium, Inc.
+
```

Comment:
Have you been authorized by Cavium to make this contribution on their behalf?

Reply:
I don't understand. Does the acknowledgements contain a list of contributors or a list of companies who developed the submitted code ?

Reply:
@danh-arm The firmware was open source by cavium on github and released under BSD license. So it should be fine

Reply:
This file should contain the companies that contributed if the contributor is an employee of that company, or the individual names of contributors that do not work for a particular company. In your case, it sounds like you are trying to upstream code from another GitHub project without Cavium's consent or knowledge. If that is the case, then it is your company (9Elements) that should be listed here, not Cavium. Not only that but we need to assess the provenance of that code and the value of it being in the upstream. If no-one is prepared to maintain this code, then it might as well remain out of tree. Can you provide a link to the source GitHub project?

Reply:
@danh-arm and @PatrickRudolph I am from Cavium and we are not aware of this upstreaming support of OcteonTx. And I have confirmed that no one from Cavium OcteonTx group has authorized this activity. Since the original authors of this whole code has been modified it will not be Cavium's responsiblity to fix any bugs in this code and maintain it in future. 
Also we have chip errata that is not public and so 9elements could land in trouble. I have forwarded this to our legal team and they will take appropriate action.

Reply:
We never modified any license headers. All rights belong to Cavium Inc. We just upstreamed the already public available code under BSD license

Reply:
@rchintakuntla FYI, the code is from the Octeon-TX SDK release here: https://github.com/Cavium-Open-Source-Distributions/OCTEON-TX-SDK

It will be mutually beneficial for Cavium and their customers to have ATF support upstreamed. I'd like to discuss this further with management to see if we can get resources devoted to this effort. Working with the unsupported, out-of-tree ATF from the tarball on Github is sub-optimal for everybody (especially on our end).

Reply:
@dhendrix and @zaolin  having code on github doesn't authorize anyone to upstream code. Since ATF is opensource we publish it on github along with lot other stuff. It is purely a FB and 9Elements thing now and Cavium has no role in this. Please remove Cavium Inc from maintainers atleast so that we will not be held responsible for maintaining. This is old piece of code anyway.
I am not sure what will fB gain from upstreaming this. None of our customers are going to benefit from this as they modify a lot on what we provide.

Reply:
@rchintakuntla I'll set up a meeting with my contacts at Cavium to discuss this further.

It is unexpected to run into this kind of resistance to upstreaming open-source code to support Cavium products. Those present in previous meetings seemed onboard and Cavium is active in upstream Linux kernel development. But if Cavium doesn't want to participate in upstream ATF development then we'll just work off a fork like everyone else; I don't see any technical benefits to that approach but it is what it is.

Reply:
@dhendrix you are misinterpreting my statements. What I meant is: if FB wants to upstream Cavium's old ATF code for whatever reason they seemed fit they can go ahead but please DO NOT PUT Cavium as maintainer as it totally seems pointless. It's an old code and I do not see a benefit. Also the authorship has been hijacked which I am totally against. Put maintainer as either facebook or 9elements then Cavium can be off the hook. This is the only reason I responded as @danh-arm pointed this as well. I don't know why you think this is resistance. Maybe you are new to upstreaming ? 

Had it been FB requesting Cavium actual patchset against current ATF code base then that would be a meaningful approach and then can expect some participation from Cavium.

BTW what Cavium upstreams and what it doesn't depends on Cavium's business requirements and resources to do that. For one example, if we upstream our ATF I do not see a point where some individual associated with Cavium as a customer is going to benefit. There are lot of pieces missing plus we do lot of custom modifications to satisfy some customer requirements. Now upstreaming these will be a pain and might break other platforms. Linux kernel is a different ball game and there's a very good reason for that. I am sure you know that.